service/root.go

```diff
@@ -0,0 +1,60 @@
+package service
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+	"github.com/riandyrn/otelchi"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/zenledger-io/zazen/httpx"
+	"github.com/zenledger-io/zazen/telemetry"
+)
+
+var (
+	defaultRequestTimeout = 60 * time.Second
+)
+
+func newRoot(tp trace.TracerProvider, cfg Config) http.Handler {
+	root := chi.NewRouter()
+
+	timeout := cfg.RequestTimeout
+	if timeout == 0 {
+		timeout = defaultRequestTimeout
+	}
+
+	root.Use(middleware.Timeout(timeout))
+	root.Use(middleware.RealIP)
```

Comment:
Does this handle IP from cloudflare?

Reply:
It handles all the reverse-proxy headers I know of: https://github.com/go-chi/chi/blob/master/middleware/realip.go#L12-L14